apps/emqx/include/asserts.hrl

```diff
@@ -83,6 +83,27 @@
     end)()
 ).
 
+-define(assertMatchOneOf(PAT1, PAT2, EXPR),
```

Comment:
Nit: this macro looks very situational and too complicated for its own good. 
I would prefer to just have a simple case experession throwing badmatch in the test code.

Reply:
I see the point, but

> The benefit of using macros like `assertMatch` is that everyone knows their semantics. It's not the case with a new macro: anyone who reads or modifies the code will end up looking up its definition...

This honestly reads like a convention that will effectively force us to not introduce any macros whatsoever anymore. Yet I don't think we really want to adopt such convention.

I decided to introduce a new macro because it isn't nearly the first time that I need something like this in the test code (probably third or fourth). Anyway, I'll cut it out of this PR, agree that it isn't too important.

Reply:
Well, there's value in following KISS principle too. 
This particular macro caught my attention, because it's called `assertMatchOneOf`, but it handles only two cases. So on one hand it pretends to be generic, but in reality it is limited. What if someone needs to match one of three cases? Create a new macro? Write explicit match, and have it side-by-side with `assertMatchOneOf`?